GetJobber/hash_accessor
# JOB-12660 ruby 2.4 compatibility

# What's in this PR?

This removes a ruby 2.4 deprecation about `Fixnum`, so it's not so noisy when you're using it.

Additionally, this adds a note saying this entire gem is deprecated. No one outside of our organization should be using it going forward.

# QA

To test this, have ruby 2.4 installed:
- `rvm install ruby-2.4.5`
- `rvm use 2.4.5`

To see the "before" behavior, change the instance of `Integer` back to `Fixnum` and run these new tests with:
- `rake test`

With `Fixnum` in the class method, you'll see a deprecation warning from the new tests. With `Integer`, you won't see a deprecation warning.

To confirm that it still works with ruby 2.3, switch back to that ruby with rvm:
- `rvm use 2.3.8`

Then run the tests again.

If you're paranoid about what ruby version is being used by the tests, you can add in a `puts RUBY_VERSION` into the test temporarily.
